<a href="https://colab.research.google.com/github/Chhavinder058/CAPSTONE/blob/main/Capstione_Hypothesis__Statements.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **CAPSTONE - Farming 2.0: IoT and Machine Learning Integration**

## **Team Members**

1.Chhavinder Singh - 0830731

2.Gireeshma Yarradoddi - 0858771

3.Osama Al Daqaq - 0847700

4.Santosh Gopi - 0850689

5.Amandeep Kaur - 0837476

# **HYPOTHESIS STATEMENTS**

In [15]:
# Hypothesis 1: Regions with higher rainfall have higher crop yields.
hypothesis_1 = "Higher Rainfall_mm is associated with higher Yield_tons_per_hectare."

# Hypothesis 2: The use of fertilizers leads to a significant increase in crop yield.
hypothesis_2 = "Fertilizer_Used (True) has a higher Yield_tons_per_hectare compared to when it is not used (False)."

# Hypothesis 3: Temperature impacts the number of days to harvest.
hypothesis_3 = "Higher Temperature_Celsius is correlated with a shorter Days_to_Harvest."

In [14]:
#Importing all the necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score


In [4]:
#Loading the dataset
from google.colab import drive
drive.mount('/content/drive')
data = pd.read_csv("crop_yield.csv")

# Display the first 5 rows of the data
print(data.head())

Mounted at /content/drive
  Region Soil_Type     Crop  Rainfall_mm  Temperature_Celsius Fertilizer_Used  \
0   West     Sandy   Cotton   897.077239            27.676966           False   
1  South      Clay     Rice   992.673282            18.026142            True   
2  North      Loam   Barley   147.998025            29.794042           False   
3  North     Sandy  Soybean   986.866331            16.644190           False   
4  South      Silt    Wheat   730.379174            31.620687            True   

  Irrigation_Used Weather_Condition  Days_to_Harvest  Yield_tons_per_hectare  
0            True            Cloudy            122.0                6.555816  
1            True             Rainy            140.0                8.527341  
2           False             Sunny            106.0                1.127443  
3            True             Rainy            146.0                6.517573  
4            True            Cloudy            110.0                7.248251  


In [5]:
data.describe() #Provides the statistical summary

Rainfall_mm  Temperature_Celsius  Days_to_Harvest  \
count  112260.000000        112260.000000    112259.000000   
mean      548.644859            27.504382       104.416029   
std       259.706819             7.218918        25.979104   
min       100.006823             3.000000        60.000000   
25%       323.431830            21.251081        82.000000   
50%       548.658027            27.545796       104.000000   
75%       772.380458            33.761707       127.000000   
max       999.996116            39.998947       149.000000   

       Yield_tons_per_hectare  
count           112259.000000  
mean                 4.643501  
std                  1.696499  
min                 -0.841617  
25%                  3.413154  
50%                  4.649180  
75%                  5.867159  
max                  9.727004

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112260 entries, 0 to 112259
Data columns (total 10 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Region                  112260 non-null  object 
 1   Soil_Type               112260 non-null  object 
 2   Crop                    112260 non-null  object 
 3   Rainfall_mm             112260 non-null  float64
 4   Temperature_Celsius     112260 non-null  float64
 5   Fertilizer_Used         112259 non-null  object 
 6   Irrigation_Used         112259 non-null  object 
 7   Weather_Condition       112259 non-null  object 
 8   Days_to_Harvest         112259 non-null  float64
 9   Yield_tons_per_hectare  112259 non-null  float64
dtypes: float64(4), object(6)
memory usage: 8.6+ MB


In [10]:
# Checks for missing values in each column
missing_values = data.isnull().sum()

# Displays the count of missing value.
print("Missing values in each column:")
print(missing_values)

# It checks the only columns with missing values:
missing_columns = missing_values[missing_values > 0]
print("\nColumns with missing values:")
print(missing_columns)

Missing values in each column:
Region                    0
Soil_Type                 0
Crop                      0
Rainfall_mm               0
Temperature_Celsius       0
Fertilizer_Used           1
Irrigation_Used           1
Weather_Condition         1
Days_to_Harvest           1
Yield_tons_per_hectare    1
dtype: int64

Columns with missing values:
Fertilizer_Used           1
Irrigation_Used           1
Weather_Condition         1
Days_to_Harvest           1
Yield_tons_per_hectare    1
dtype: int64


In [13]:
# Fill missing values for numeric columns with their mean
data['Days_to_Harvest'] = data['Days_to_Harvest'].fillna(data['Days_to_Harvest'].mean())
data['Yield_tons_per_hectare'] = data['Yield_tons_per_hectare'].fillna(data['Yield_tons_per_hectare'].mean())

# Fill missing values for categorical columns with the mode (most frequent value)
data['Fertilizer_Used'] = data['Fertilizer_Used'].fillna(data['Fertilizer_Used'].mode()[0])
data['Irrigation_Used'] = data['Irrigation_Used'].fillna(data['Irrigation_Used'].mode()[0])
data['Weather_Condition'] = data['Weather_Condition'].fillna(data['Weather_Condition'].mode()[0])

# Verify that there are no missing values
print("Missing values after filling:")
print(data.isnull().sum())

Missing values after filling:
Region                    0
Soil_Type                 0
Crop                      0
Rainfall_mm               0
Temperature_Celsius       0
Fertilizer_Used           0
Irrigation_Used           0
Weather_Condition         0
Days_to_Harvest           0
Yield_tons_per_hectare    0
dtype: int64


# **Hypothesis 1: Regions with higher rainfall have higher crop yields.**
hypothesis_1 = "Higher Rainfall_mm is associated with higher Yield_tons_per_hectare."

In [16]:
import scipy.stats as stats

# Calculate the correlation between Rainfall_mm and Yield_tons_per_hectare
correlation, p_value = stats.pearsonr(data['Rainfall_mm'], data['Yield_tons_per_hectare'])

print(f"Correlation: {correlation}")
print(f"P-value: {p_value}")

# Interpretation: If p_value < 0.05, there is a statistically significant correlation between rainfall and crop yield.

Correlation: 0.7632092307798274
P-value: 0.0


## **Hypothesis Explaination:**
1. Correlation (0.763): The correlation value is 0.763, which is close to 1. This means there is a strong positive relationship between rainfall and crop yield. As rainfall increases, crop yield tends to increase as well.
2. P-value (0.0): The p-value is 0.0, which is less than 0.05. This indicates that the relationship between rainfall and crop yield is statistically significant and unlikely to be due to random chance.

# **Conclusion**:

The data strongly supports our hypothesis that regions with higher rainfall tend to have higher crop yields. This is backed by both the strong correlation and the significant p-value.

# **Hypothesis 2: The use of fertilizers leads to a significant increase in crop yield.**
hypothesis_2 = "Fertilizer_Used (True) has a higher Yield_tons_per_hectare compared to when it is not used (False)."

In [17]:
# Separate the data into two groups based on Fertilizer_Used
fertilizer_used = data[data['Fertilizer_Used'] == True]['Yield_tons_per_hectare']
fertilizer_not_used = data[data['Fertilizer_Used'] == False]['Yield_tons_per_hectare']

# Perform an independent t-test
t_stat, p_value = stats.ttest_ind(fertilizer_used, fertilizer_not_used)

print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Interpretation:
# If p_value < 0.05, there is a significant difference in yields based on fertilizer use.

T-statistic: 167.34041111890866
P-value: 0.0


## **Hypothesis Explanation**
1.	T-statistic (167.34): The t-statistic measures the size of the difference between the average crop yields with and without fertilizer. A very large t-statistic, like 167.34, means that the difference in yields between the two groups is very significant.

2.	P-value (0.0): The p-value is 0.0, which is much smaller than 0.05. This means the difference in crop yields between using fertilizer and not using fertilizer is statistically significant and not due to random chance.

# **Conclusion**:

The data strongly supports our hypothesis that using fertilizer significantly increases crop yields. The results show a clear and meaningful difference in yield between crops grown with fertilizer and those grown without it.

# **Hypothesis 3: Temperature impacts the number of days to harvest.**
hypothesis_3 = "Higher Temperature_Celsius is correlated with a shorter Days_to_Harvest."

In [18]:
# Calculate the correlation between Temperature_Celsius and Days_to_Harvest
correlation, p_value = stats.pearsonr(data['Temperature_Celsius'], data['Days_to_Harvest'])

print(f"Correlation: {correlation}")
print(f"P-value: {p_value}")

# Interpretation:
# If p_value < 0.05, there is a statistically significant correlation between temperature and time to harvest.

Correlation: 0.0022070093755452106
P-value: 0.4596300937067529


## **Third hypothesis:**

1.	Correlation (0.0022): The correlation value is 0.0022, which is extremely close to 0. This indicates that there is no meaningful relationship between temperature and the number of days to harvest.

2.	P-value (0.4596): The p-value is 0.4596, which is much greater than 0.05. This means the result is not statistically significant, and any small correlation observed is likely due to random chance.

## **Conclusion**:

The data does not support our hypothesis that temperature significantly impacts the number of days to harvest. There appears to be no meaningful connection between these two variables based on this analysis.